org.uqbar.project.wollok.typeSystem/src/org/uqbar/project/wollok/typesystem/constraints/ConstraintBasedTypeSystem.xtend

```diff
@@ -176,23 +184,45 @@ class ConstraintBasedTypeSystem implements TypeSystem, TypeProvider {
 	}
 
 	def classType(WClass clazz) {
-		new ClassBasedWollokType(clazz, this)
+		genericTypes.get(clazz.fqn) ?: new ClassBasedWollokType(clazz, this)
```

Comment:
Note that classType can return a genericType, if the received class is known to be generic. All generic types should be known before arriving here!